core/src/main/java/org/elasticsearch/script/ScriptMetaData.java

```diff
@@ -70,14 +70,15 @@ public String getScript(String language, String id) {
         return scriptAsBytes.utf8ToString();
     }
 
-    public static String parseStoredScript(BytesReference scriptAsBytes) {
+    public static String parseStoredScript(BytesReference scriptAsBytes, XContentType xContentType) {
         // Scripts can be stored via API in several ways:
         // 1) wrapped into a 'script' json object or field
         // 2) wrapped into a 'template' json object or field
         // 3) just as is
         // In order to fetch the actual script in consistent manner this parsing logic is needed:
         // EMPTY is ok here because we never call namedObject, we're just copying structure.
-        try (XContentParser parser = XContentHelper.createParser(NamedXContentRegistry.EMPTY, scriptAsBytes);
+        try (XContentParser parser = xContentType == null ? XContentHelper.createParser(NamedXContentRegistry.EMPTY, scriptAsBytes) :
```

Comment:
can we add a comment here about null  content types? that is for bw comp given that we may have scripts already stored in different content types? and this has to stay in master too?